source/server/connection_handler_impl.cc

```diff
@@ -42,6 +42,14 @@ void ConnectionHandlerImpl::addListener(absl::optional<uint64_t> overridden_list
 
   auto details = std::make_shared<ActiveListenerDetails>();
   if (config.internalListenerConfig().has_value()) {
+    // Ensure the this ConnectionHandlerImpl link to the thread local registry. Ideally this step
+    // should be done only once. However, an extra phase and interface is overkill.
+    Network::InternalListenerRegistry& internal_listener_registry =
```

Comment:
Will this test work if internal listener extension is disabled?

Reply:
Connection Handler doesn't have the flow to fail the addListener().
This `config.internalListenerConfig().has_value()` branch should be handled by ListenerManagerImpl which would reject such config on LDS update.

Reply:
Confirmed by deleting source/extensions/bootstrap/internal_listener and the item in source/extensions/extensions_build_config.bzl

All tests passed